x-pack/plugins/notifications/server/types.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EmailService, EmailServiceStartDeps } from './services';
+
+export type NotificationsPluginStartDeps = EmailServiceStartDeps;
+
+export interface NotificationsPluginStart {
+  email?: EmailService;
```

Comment:
The `Notifications` plugin no longer exposes a `Promise<EmailService>`, but the email service directly (only available in `start` contract).

Should any errors occur during initialization (e.g. default connector not preconfigured), the `NotificationsPlugin` will log a `warn` and the `email` property will be `undefined`.

Reply:
Overall switching from async to sync access seems like a better DX. 

I wonder if we couldn't find a better way to handle the case where the email service couldn't be created (for any reason). Just having the `email` property on the contract be `undefined` feels kinda wrong to me...